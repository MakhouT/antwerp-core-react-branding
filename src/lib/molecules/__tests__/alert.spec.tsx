import * as React from "react";
import * as renderer from "react-test-renderer";

import * as A from "../..";

describe("Alert", () => {
    it("Renders default", () => {
        // Act
        const component: any = renderer.create(
            <A.Alert
                message="Message"
                title="Jest Testing"
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders visible", () => {
        // Act
        const component: any = renderer.create(
            <A.Alert
                message="Message"
                title="Jest Testing"
                visible
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders visible, primary", () => {
        // Act
        const component: any = renderer.create(
            <A.Alert
                message="Message"
                title="Jest Testing"
                visible
                level={A.Levels.Primary}
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders visible, warning", () => {
        // Act
        const component: any = renderer.create(
            <A.Alert
                message="Message"
                title="Jest Testing"
                visible
                level={A.Levels.Warning}
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders visible, danger", () => {
        // Act
        const component: any = renderer.create(
            <A.Alert
                message="Message"
                title="Jest Testing"
                visible
                level={A.Levels.Danger}
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders visible, danger, with onOk", () => {
        // Act
        const component: any = renderer.create(
            <A.Alert
                message="Message"
                title="Jest Testing"
                visible
                level={A.Levels.Danger}
                onOk={() => { /**/ }}
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });

    it("Renders visible, danger, with onOk and onCancel", () => {
        // Act
        const component: any = renderer.create(
            <A.Alert
                message="Message"
                title="Jest Testing"
                visible
                level={A.Levels.Danger}
                onOk={() => { /**/ }}
                onCancel={() => { /**/ }}
            />);
        const tree: renderer.ReactTestRendererJSON = component.toJSON();

        // Assert
        expect(tree).toMatchSnapshot();
    });
});
