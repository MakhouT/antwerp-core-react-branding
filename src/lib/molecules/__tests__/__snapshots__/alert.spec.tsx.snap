// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Renders default 1`] = `null`;

exports[`Alert Renders visible 1`] = `
<div
  className="a-overlay a-overlay--dark"
>
  <div
    className="m-alert"
  >
    <h5
      className="u-margin-bottom-xs"
    >
      Jest Testing
    </h5>
    <p
      className="u-margin-bottom"
    >
      Message
    </p>
    <div
      className="m-alert__actions"
    >
      <button
        autoFocus={true}
        className="a-button a-button--default a-button--small m-alert__ok"
        disabled={undefined}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        type="submit"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`Alert Renders visible, danger 1`] = `
<div
  className="a-overlay a-overlay--dark"
>
  <div
    className="m-alert m-alert--danger"
  >
    <h5
      className="u-margin-bottom-xs"
    >
      Jest Testing
    </h5>
    <p
      className="u-margin-bottom"
    >
      Message
    </p>
    <div
      className="m-alert__actions"
    >
      <button
        autoFocus={true}
        className="a-button a-button--default a-button--danger a-button--small m-alert__ok"
        disabled={undefined}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        type="submit"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`Alert Renders visible, danger, with onOk 1`] = `
<div
  className="a-overlay a-overlay--dark"
>
  <div
    className="m-alert m-alert--danger"
  >
    <h5
      className="u-margin-bottom-xs"
    >
      Jest Testing
    </h5>
    <p
      className="u-margin-bottom"
    >
      Message
    </p>
    <div
      className="m-alert__actions"
    >
      <button
        autoFocus={true}
        className="a-button a-button--default a-button--danger a-button--small m-alert__ok"
        disabled={undefined}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        type="submit"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`Alert Renders visible, danger, with onOk and onCancel 1`] = `
<div
  className="a-overlay a-overlay--dark"
>
  <div
    className="m-alert m-alert--danger"
  >
    <button
      autoFocus={true}
      className="a-button-transparent a-button--danger m-alert__close has-icon"
      disabled={undefined}
      onClick={[Function]}
      style={undefined}
      tabIndex={undefined}
      type="button"
    >
      <i
        className="fa fa-close"
      />
    </button>
    <h5
      className="u-margin-bottom-xs"
    >
      Jest Testing
    </h5>
    <p
      className="u-margin-bottom"
    >
      Message
    </p>
    <div
      className="m-alert__actions"
    >
      <button
        autoFocus={true}
        className="a-button-outline a-button--danger a-button--small m-alert__cancel"
        disabled={undefined}
        onClick={[Function]}
        style={undefined}
        tabIndex={undefined}
        type="button"
      >
        Cancel
      </button>
      <button
        autoFocus={true}
        className="a-button a-button--default a-button--danger a-button--small m-alert__ok"
        disabled={undefined}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        type="submit"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`Alert Renders visible, primary 1`] = `
<div
  className="a-overlay a-overlay--dark"
>
  <div
    className="m-alert m-alert--primary"
  >
    <h5
      className="u-margin-bottom-xs"
    >
      Jest Testing
    </h5>
    <p
      className="u-margin-bottom"
    >
      Message
    </p>
    <div
      className="m-alert__actions"
    >
      <button
        autoFocus={true}
        className="a-button a-button--default a-button--primary a-button--small m-alert__ok"
        disabled={undefined}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        type="submit"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;

exports[`Alert Renders visible, warning 1`] = `
<div
  className="a-overlay a-overlay--dark"
>
  <div
    className="m-alert m-alert--warning"
  >
    <h5
      className="u-margin-bottom-xs"
    >
      Jest Testing
    </h5>
    <p
      className="u-margin-bottom"
    >
      Message
    </p>
    <div
      className="m-alert__actions"
    >
      <button
        autoFocus={true}
        className="a-button a-button--default a-button--warning a-button--small m-alert__ok"
        disabled={undefined}
        onClick={[Function]}
        style={undefined}
        tabIndex={0}
        type="submit"
      >
        OK
      </button>
    </div>
  </div>
</div>
`;
